webapp-django/crashstats/manage/views.py

```diff
@@ -508,11 +508,60 @@ def graphics_devices_lookup(request):
 def symbols_uploads(request):
     context = {}
     context['page_title'] = "Symbols Uploads"
-    context['all_uploads'] = (
+    return render(request, 'manage/symbols_uploads.html', context)
+
+
+@superuser_required
+@json_view
+def symbols_uploads_data(request):
+    try:
+        page = int(request.GET.get('page', 1))
+        assert page >= 1
+    except (ValueError, AssertionError):
+        return http.HttpResponseBadRequest('invalid page')
+
+    form = forms.FilterSymbolsUploadsForm(request.GET)
+    if not form.is_valid():
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    uploads = (
         SymbolsUpload.objects.all()
+        .select_related('user')
         .order_by('-created')
     )
-    return render(request, 'manage/symbols_uploads.html', context)
+    if form.cleaned_data['email']:
+        uploads = uploads.filter(
+            user__email__icontains=form.cleaned_data['email']
+        )
+    if form.cleaned_data['filename']:
+        uploads = uploads.filter(
+            filename__icontains=form.cleaned_data['filename']
+        )
+
+    count = uploads.count()
+    items = []
+    batch_size = settings.SYMBOLS_UPLOADS_ADMIN_BATCH_SIZE
+    m = (page - 1) * batch_size
+    n = page * batch_size
+    for upload in uploads[m:n]:
+        items.append({
+            'user': {
+                'email': upload.user.email,
+                'id': upload.user.pk,
+            },
+            'id': upload.pk,
+            'filename': upload.filename,
+            'size': upload.size,
+            'created': upload.created,
+        })
+    return {
+        'items': items,
+        'count': count,
+        'batch_size': batch_size,
+        'page': page,
+        'user_edit_url': reverse('manage:user', args=('0',)),
+        'symbols_content_url': reverse('symbols:content', args=('0',)),
```

Comment:
Not really important, but in terms of API design, I think it's better to send those URLs along with the content and without the fake ``0``. It makes the data slightly bigger (but with 10 results, does that even matter? ) but it makes the front-end code and the API results simpler to understand. 

And if you do that, you can even remove the identifiers entirely, since I don't think you use them for anything else that building URLs.